lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -361,12 +361,12 @@ void constraints::performSyntacticDiagnosticsForTarget(
 
   case SyntacticElementTarget::Kind::forEachStmt: {
     auto *stmt = target.getAsForEachStmt();
-
-    // First emit diagnostics for the main expression.
-    performSyntacticExprDiagnostics(stmt->getTypeCheckedSequence(), dc,
-                                    isExprStmt,
-                                    disableExprAvailabilityChecking);
-
+    if (auto seq = stmt->getTypeCheckedSequence()) {
```

Comment:
This goes against the pattern somewhat because previously if there was no type-checked sequence the for would be malformed, why not just return `PackExpansionExpr *`? This place should use `getTypeCheckedSequence` as well https://github.com/apple/swift/pull/67594/files#diff-8a91f0c5e88681ee619dc630205fae9083806a83c59a31b6ba35ab0ec9757a0bR1220